src/language.h

```diff
@@ -4,52 +4,53 @@
  * the Server Side Public License v1 (SSPLv1).
  */
 
-#pragma once
-#include <stdlib.h>
-
-#ifdef __cplusplus
-extern "C" {
-#endif
-
-typedef enum {
-  RS_LANG_ENGLISH = 0,
-  RS_LANG_ARABIC,
-  RS_LANG_BASQUE,
-  RS_LANG_CATALAN,
-  RS_LANG_CHINESE,
-  RS_LANG_DANISH,
-  RS_LANG_DUTCH,
-  RS_LANG_FINNISH,
-  RS_LANG_FRENCH,
-  RS_LANG_GERMAN,
-  RS_LANG_GREEK,
-  RS_LANG_HINDI,
-  RS_LANG_HUNGARIAN,
-  RS_LANG_ITALIAN,
-  RS_LANG_INDONESIAN,
-  RS_LANG_IRISH,
-  RS_LANG_LITHUANIAN,
-  RS_LANG_NEPALI,
-  RS_LANG_NORWEGIAN,
-  RS_LANG_PORTUGUESE,
-  RS_LANG_ROMANIAN,
-  RS_LANG_RUSSIAN,
-  RS_LANG_SPANISH,
-  RS_LANG_SWEDISH,
-  RS_LANG_TAMIL,
-  RS_LANG_TURKISH,
-  RS_LANG_ARMENIAN,
-  RS_LANG_SERBIAN,
-  RS_LANG_YIDDISH,
-  RS_LANG_UNSUPPORTED
-} RSLanguage;
-
-#define DEFAULT_LANGUAGE RS_LANG_ENGLISH
-
-/* check if a language is supported by our stemmers */
-RSLanguage RSLanguage_Find(const char *language, size_t len);
-const char *RSLanguage_ToString(RSLanguage language);
-
-#ifdef __cplusplus
-}
-#endif
+#pragma once
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum {
+  RS_LANG_ENGLISH = 0,
+  RS_LANG_ARABIC,
+  RS_LANG_BASQUE,
+  RS_LANG_CATALAN,
+  RS_LANG_CHINESE,
+  RS_LANG_DANISH,
+  RS_LANG_DUTCH,
+  RS_LANG_FINNISH,
+  RS_LANG_FRENCH,
+  RS_LANG_GERMAN,
+  RS_LANG_GREEK,
+  RS_LANG_HINDI,
+  RS_LANG_HUNGARIAN,
+  RS_LANG_ITALIAN,
+  RS_LANG_INDONESIAN,
+  RS_LANG_IRISH,
+  RS_LANG_LITHUANIAN,
+  RS_LANG_NEPALI,
+  RS_LANG_NORWEGIAN,
+  RS_LANG_PORTUGUESE,
+  RS_LANG_ROMANIAN,
+  RS_LANG_RUSSIAN,
+  RS_LANG_SPANISH,
+  RS_LANG_SWEDISH,
+  RS_LANG_TAMIL,
+  RS_LANG_TURKISH,
+  RS_LANG_ARMENIAN,
+  RS_LANG_SERBIAN,
+  RS_LANG_YIDDISH,
+  RS_LANG_UNSUPPORTED,
+  RS_LANG_INDEX_DEFAULT, // The user did not set the language for FT.SEARCH, use the index language

```

Comment:
Consider a neutral value such as `RS_LANG_UNSET` etc., that will keep the logic outside of the enum.

Reply:
Need to double check the new value doesn't get to `SchemaRule_RdbSave`

Reply:
> Consider a neutral value such as RS_LANG_UNSET etc., that will keep the logic outside of the enum.

A neutral value is now used: `RS_LANG_UNSET`

> Need to double check the new value doesn't get to `SchemaRule_RdbSave`

These are the validations to check that the value `RS_LANG_UNSET` does not get to `SchemaRule_RdbSave`:
1. The value `RS_LANG_UNSET` is only assigned to the variable used for the search `RSSearchOptions->language`, it is not assigned to schema rule variable
2. The schema rule variable ``SchemaRule->lang_default` receives the result of the function `RSLanguage_Find()`, but that functions does not return `RS_LANG_UNSET`
3. `RSSearchOptions->language` is not assigned to `SchemaRule->lang_default`